crates/ark/src/modules/positron/methods.R

```diff
@@ -24,7 +24,7 @@ ark_methods_table$ark_positron_variable_get_children <- new.env(
 )
 lockEnvironment(ark_methods_table, TRUE)
 
-ark_methods_allowed_packages <- c("torch", "reticulate")
+ark_methods_allowed_packages <- c("torch", "reticulate", "duckplyr")
```

Comment:
I wonder if we also want to bless duckdb, this is where the problematic objects ultimately come from.

```suggestion
ark_methods_allowed_packages <- c("torch", "reticulate", "duckplyr", "duckdb")
```

Reply:
But duckplyr owns `duckplyr_df` and that's what we'd register methods for.

Its still WIP but duckplyr would gain a `positron.R` file that looks somewhat like this

(see also https://github.com/mlverse/torch/blob/ca61df853546f6e2248f5de718d0f9d9c7163a8c/R/positron.R#L1)

```r
ARK_REGISTER_FUNCTION <- ".ark.register_method"

positron_is_enabled_to_register <- function() {
  exists(ARK_REGISTER_FUNCTION, mode = "function")
}

register_ark_method <- function(generic, class, method) {
  register_call <- call(
    ARK_REGISTER_FUNCTION,
    quote(generic),
    quote(class),
    quote(method)
  )
  eval(register_call, envir = environment())
}

register_positron_methods <- function() {
  if (!positron_is_enabled_to_register()) {
    return()
  }

  tryCatch(
    register_positron_methods_impl(),
    error = function(err) {
      cli::cli_warn("Failed to register Positron methods.", parent = err)
    }
  )
}

register_positron_methods_impl <- function() {
  register_ark_method(
    "ark_positron_variable_display_value",
    "duckplyr_df",
    function(x, ...) {
      duckplyr_df_variable_display_value(x)
    }
  )

  register_ark_method(
    "ark_positron_variable_display_type",
    "duckplyr_df",
    function(x, ...) {
      duckplyr_df_variable_display_type(x)
    }
  )
}

duckplyr_df_variable_display_value <- function(x) {
  n_col <- df_n_col(x)

  if (n_col == 1L) {
    col_word <- "column"
  } else {
    col_word <- "columns"
  }

  sprintf("[? rows x %s %s]", n_col, col_word)
}

duckplyr_df_variable_display_type <- function(x) {
  "duckplyr_df"
}
```

and in your onLoad:

```r
.onLoad <- function(lib, pkg) {
  run_on_load()
  register_positron_methods()
}
```

These methods are registered with Ark when duckplyr is loaded, and Ark uses them when communicating with Positron about `duckplyr_df` objects to avoid the code paths where we are otherwise forced to try and compute the number of rows.